lib/runner/index.js

```diff
@@ -90,26 +103,43 @@ _.assign(Runner, {
     /**
      * @private
      *
-     * @param {Collection} collection
-     * @param {String} groupId
-     * @returns {Array}
+     * @param {?Collection} collection
+     * @param {?String} groupId
+     * @param {Function} callback
+     * @returns {Object}
+     * @todo Add support for accepting groupId as an array of ids/names
      */
-    extractRunnableItems: function (collection, groupId) {
-        var items = [],
+    extractRunnableItems: function (collection, groupId, callback) {
+        _.isFunction(groupId) && (callback = groupId, groupId = null);
+        _.isFunction(collection) && (callback = collection, groupId = null, collection = null);
+
+        if (!Collection.isCollection(collection)) {
+            return callback(new Error('A valid collection SDK instance is required'));
+        }
+
+        var idMatch,
+            nameMatch,
+            items = [],
             group = !groupId && collection; // if a group id is unspecified, the collection is our group
 
         // if group id is provided, we try and check if we can find it in the collection
-        !group && collection.forEachItemGroup(function (itemGroup) {
-            // @todo: fix this. the check by id should all finish before doing check by name
-            (itemGroup.id === groupId || itemGroup.name === groupId) && (group = itemGroup);
-        });
+        if (!group) {
+            collection && collection.forEachItemGroup && collection.forEachItemGroup(function (itemGroup) {
```

Comment:
redundant if collection is typechecked earlier

Reply:
Addressed